core/sql/executor/ExHdfsScan.cpp

```diff
@@ -1700,6 +1714,138 @@ char * ExHdfsScanTcb::extractAndTransformAsciiSourceToSqlRow(int &err,
   return NULL;
 }
 
+void ExHdfsScanTcb::computeRangesAtRuntime()
```

Comment:
volatile table create is very slow, IMHO, it is unacceptable for a general usage. A single run of create volatile table is about 10 seconds. But I can work on it .